packages/salesforcedx-apex-replay-debugger/test/unit/core/logContext.test.ts

```diff
@@ -183,6 +183,29 @@ describe('LogContext', () => {
     expect(printToDebugConsoleStub.calledTwice).to.be.true;
   });
 
+  it('Should detect and parse HEAP_DUMP log entries', () => {
+    readLogFileStub.restore();
+    readLogFileStub = sinon
+      .stub(LogContextUtil.prototype, 'readLogFile')
+      .returns([
+        '43.0 APEX_CODE,FINEST;...;VISUALFORCE,FINER;..',
+        '<TimeInfo>|HEAP_DUMP|[11]|<HeapDumpId1>|<ClassName1>|<Namespace1>|11',
+        '<TimeInfo>|HEAP_DUMP|[22]|<HeapDumpId2>|<ClassName2>|<Namespace2>|22'
```

Comment:
You were looking into if these two line numbers can ever be different. Any findings from that?

Reply:
Long story short, I don't see a way where the two line numbers can be different that we'd actually take the heap_dump. The second one is from the checkpoint, itself, and that's the one I feel is safest to use because we know, very specifically, where it comes from.